nomad/data_source_jwks.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"math/big"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/id"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceJWKS() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceJWKSRead,
+		Schema: map[string]*schema.Schema{
+			"keys": {
+				Description: "JSON Web Key Set (JWKS) public keys for validating workload identity JWTs",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key_use": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"key_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"key_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"algorithm": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"modulus": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"exponent": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"pem_keys": {
+				Description: "JWKS as a list of PEM keys",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+			},
+		},
+	}
+}
+
+type Key struct {
+	KeyUse    string `json:"use"`
+	KeyType   string `json:"kty"`
+	KeyId     string `json:"kid"`
+	Algorithm string `json:"alg"`
+	Modulus   string `json:"n"`
+	Exponent  string `json:"e"`
+}
+
+func dataSourceJWKSRead(d *schema.ResourceData, meta any) error {
+	client := meta.(ProviderConfig).client
+	operator := client.Raw()
+	queryOpts := &api.QueryOptions{}
+
+	jwks := struct {
+		Keys []Key `json:"keys"`
+	}{}
+
+	log.Printf("[DEBUG] Reading JWKS from Nomad")
+	_, err := operator.Query("/.well-known/jwks.json", &jwks, queryOpts)
+
+	if err != nil {
+		return fmt.Errorf("error reading JWKS from Nomad: %s", err)
+	}
+
+	if len(jwks.Keys) == 0 {
+		return fmt.Errorf("no keys found")
+	}
+
+	d.SetId(id.UniqueId())
+	if err := d.Set("keys", fromKeys(jwks.Keys)); err != nil {
+		return fmt.Errorf("error setting JWKS: %#v", err)
+	}
+
+	pemKeys := make([]string, 0, len(jwks.Keys))
+
+	for _, key := range jwks.Keys {
+		pemKey, err := keyToPem(key)
+		if err != nil {
+			return fmt.Errorf("Could not encode JWK as PEM: %s", err)
+		}
+		pemKeys = append(pemKeys, pemKey)
+	}
+
+	if err := d.Set("pem_keys", pemKeys); err != nil {
+		return fmt.Errorf("error setting JWKS pemKeys: %s", err)
+	}
+
+	return nil
+}
+
+func keyToPem(key Key) (string, error) {
+
+	// FIXME does Nomad always use RSA keys for JWKS?
```

Comment:
```suggestion
    // Nomad also supports EdDSA keys, but they are not used for OIDC so only
    // RSA keys should be listed in this endpoint.
```

I wanted to push a quick fix to this comment, but it doesn't seem like I have write access to the PR 😅 

Here's what I was thinking of writing. There's also a conflict in `CHANGELOG.md` that needs to be fixed before we can merge it.